kernel/thread.c

```diff
@@ -248,6 +248,41 @@ int z_impl_k_thread_name_copy(k_tid_t thread_id, char *buf, size_t size)
 #endif /* CONFIG_THREAD_NAME */
 }
 
+const char *k_thread_state_str(k_tid_t thread_id)
+{
+	switch (thread_id->base.thread_state) {
+	case 0:
+		return "";
+		break;
+	case _THREAD_DUMMY:
+		return "dummy";
+		break;
+	case _THREAD_PENDING:
+		return "pending";
+		break;
+	case _THREAD_PRESTART:
+		return "restart";
+		break;
+	case _THREAD_DEAD:
+		return "dead";
+		break;
+	case _THREAD_SUSPENDED:
+		return "suspended";
+		break;
+	case _THREAD_ABORTING:
+		return "aborting";
+		break;
+	case _THREAD_QUEUED:
+		return "queued";
+		break;
+	default:
+		__ASSERT(0, "thread state (0x%02X) must be exclusive",
```

Comment:
since there is a chance that thread will be in multiple (!) states I would not assert. Just return `unknown`.

Reply:
I'm thinking that your first idea that state is exclusive seems to be valid. But right now i agree that this assert must be removed. Done

If x86 states must differ from the generic thread states
```
#define _INT_ACTIVE (1 << 7)
#define _EXC_ACTIVE (1 << 6)
```
They must become BIT(7) & BIT(8) which overfill the 8 bit value. But if we assume that states are exclusive, bitmask could be replaced with simple enum.

Reply:
current implementation of scheduler does not perform atomic transitions from one state to another, e.g. going to pending state is done in steps: remove from ready queue and unset ready state, go to pend state. In that example for a moment thread is in dummy state. I don't know if it may lead to some issues apart from logging. Maintainers have been pinged here so let them decide.